ethstorage/archiver/guide-devnet.md

```diff
@@ -0,0 +1,366 @@
+```
+
+Finally, navigate to the Optimism monorepo and locate `batchInboxAddress` in the file `packages/contracts-bedrock/deploy-config/devnetL1.json`:
+
+```json
+  "batchInboxAddress": "0xff00000000000000000000000000000000000901",
+```
+Update the value of `batchInboxAddress` with the address of the contract you just deployed. 
+
+Now, navigate to the parent directory in preparation for the next steps.
+
+### Running a Proxy of L1 Beacon
+
+For the convenience of testing, you will start a proxy for the Beacon API with a shorter blob retention period.
+
+First retrieve the beacon genesis time for later use:
+```bash
+curl -s http://localhost:5052/eth/v1/beacon/genesis | jq -r '.data.genesis_time'
+
+1732529739
+
+export GENESIS_TIME=1732529739 // replace with the actual timestamp
+```
+
+The following commands start a proxy to Beacon API with a shorter blobs retension period:
+```bash
+git clone https://github.com/ethstorage/beacon-api-wrapper.git
+cd beacon-api-wrapper
+go run cmd/main.go -b http://localhost:5052 -p 3602 -g $GENESIS_TIME -r 1200
+```
+This setup allows you to test archive service effectively. 
+For blob requests, if the request is within the latest 100 slots, the proxy will retrieve blobs from `http://localhost:5052`. For requests older than that, it will return an empty list.
+
+## Running EthStorage Node
+
+
+### Installation
+
+To set up the es-node, first clone the repository and build it:
+```bash
+git clone https://github.com/ethstorage/es-node.git
+cd es-node
+git checkout v0.1.16
+make
+```
+
+### Initialization
+
+Initialize es-node:
+```bash
+./init-rpc.sh \
+--l1.rpc http://localhost:8545 \
+--storage.l1contract $ES_CONTRACT
+```
+
+### Running ES Node in Archiver Mode
+
+Finally, run the es-node:
+
+```bash
+./run-rpc.sh \
+--storage.l1contract $ES_CONTRACT \
+--l1.rpc http://localhost:8545 \
+--l1.beacon http://localhost:5052 \
+--l1.beacon-based-time $GENESIS_TIME \
+--l1.beacon-based-slot 0 \
+--p2p.listen.udp 30375 \
+--p2p.listen.tcp 9733 \
+--rpc.port 9745 \
+--archiver.port 6678 \
+--archiver.enabled 
+```
+
+Now, navigate to the parent directory in preparation for the next steps.
+
+## L2 Setup
+
+### Running L2
+
+Once again, enter the Optimism monorepo and start the Layer 2 environment by executing the following command:
+```bash
+make devnet-up-l2
+```
+
+This command will start the following services:
+- Container ops-bedrock-l2-1
+- Container ops-bedrock-op-batcher-1
+- Container ops-bedrock-op-node-1
+- Container ops-bedrock-op-proposer-1
+- Container ops-bedrock-op-challenger-1
+- Container ops-bedrock-artifact-server-1
+
+Now, navigate to the parent directory in preparation for the next steps.
+
+The following steps will add an additional OP node in validator mode, configured to sync expired blob data from the es-node. This aims to verify that the functions of the derivation pipeline are working correctly with the BatchInbox contract and the EthStorage archive service. 
```

Comment:
Do we need to wait at least 100 slots for some blobs to expire?

Reply:
Here `OP node` is ambiguous, `OP Stack instance` may be better.

Reply:
The archiver service itself is unaware of dApps. Isn't it?